client/ayon_core/pipeline/create/creator_plugins.py

```diff
@@ -831,6 +833,59 @@ def get_pre_create_attr_defs(self):
         """
         return self.pre_create_attr_defs
 
+    def apply_staging_dir(self, instance):
```

Comment:
I don't understand what it applies? Shouldn't that be `prepare_staging_dir`?

Reply:
I would preffer if there would be method that would just "get" the staging info for given instance and second method that would "apply it" (calling the other method and add it to transient data). Right now I can't "just get" the staging dir without all the other logic.

We also should use context from instance instead of current context.

Reply:
I've splited the function in 2 as suggested.
Sorry silly question but how do you gather the context from the instance ? I cannot find the equivalent from the instance.

Reply:
I still see only `apply_staging_dir` which has a lot of logic to get the staging dir for an instance.